service/raft/raft_address_map.hh

```diff
@@ -329,10 +329,8 @@ public:
     }
 
     // A shortcut to setting a new permanent address
-    void set(raft::server_address addr) {
-        return set(addr.id,
-            ser::deserialize_from_buffer(addr.info, boost::type<gms::inet_address>{}),
-            false);
+    void set(raft::server_id id, raft::server_info info) {
+        return set(id, ser::deserialize_from_buffer(std::move(info), boost::type<gms::inet_address>{}), false);
```

Comment:
This patch is LGTM but I find it of very low value. You still keep the server_address data structure internally. You still set can_vote to false in it. Now you have to pass two arguments instead of one. Some call sites became more verbose. The goal is, as always to have *few* powerful concepts, not many specific ones. This patch just replaces one weak abstraction with another.

Reply:
> You still keep the server_address data structure internally. You still set can_vote to false in it.

Which place do you have in mind? What does the "internally" refer to? If you mean the call site of `set` - I'm removing `raft::server_address` from there in later commits.

Reply:
> You still set can_vote to false in it

There will no longer be `can_vote` after a few commits.